hkmc2/shared/src/test/mlscript/rp/MatchResult.mls

```diff
@@ -19,7 +19,11 @@ if MatchFailure("oops") is MatchFailure(x) then x
 
 data class MatchResult(x, y, z)
 
+:e
 MatchResult(0)
+//│ ╔══[ERROR] Expected 3 arguments, got 1
+//│ ║  l.23: 	MatchResult(0)
+//│ ╙──      	           ^^^
 //│ = MatchResult(0, undefined, undefined)
```

Comment:
I am not sure what is this test case testing. I suppose it should be an error?

Reply:
Yeah IDK, let's ask @chengluyu :)

Reply:
But I'm guessing it's just reprdocuing the cursed `undefined` that pop up in such situations.

Reply:
This is testing whether the compiled `unapply` method can use the correct `MatchResult` when the `MatchResult` in `Runtime` is shadowed. But I might have gotten the number of arguments wrong. You can change the parameters of the `MatchResult` above to just one.

Reply:
I fixed this.